ComputerVision.cs

```diff
@@ -0,0 +1,518 @@
+                        GUI.DrawTexture(new Rect((obj.rect[0] / imageShrinkingRatio), (obj.rect[1] / imageShrinkingRatio), width, height), obj.confidence >= confidenceThreshold / 100 ? highConfidenceTexture : lowConfidenceTexture, ScaleMode.StretchToFill, true, width / height);
+                    
+                    TextStyle.font = objectClassFont;
+                    TextStyle.normal.textColor = Color.white;
+                    TextStyle.fontSize = objectClassFontSize;
+                    string objClass = obj.@class + (showConfidenceLevel == true ? " " + obj.confidence * 100 + "%" : "");
+                    if (objectClassFont)
+                        GUI.Label(new Rect((obj.rect[0] / imageShrinkingRatio) + (width / 2) - (25), (obj.rect[1] / imageShrinkingRatio) + 10, 50, 50), new GUIContent(objClass), TextStyle);
+                }
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Shrinks the screen shot taken to the supplied TargetWidth, then encode the new texture a JPG format and returns the binary array
+    /// Shrinking happens by organizing the pixels of the source img into the new scaled texture using the normalized UV map
+    /// </summary>
+    /// <param name="source">Screen Shot Texture</param>
+    /// <param name="targetWidth"></param>
+    /// <returns> shrank image binary </returns>
+    byte[] ShrinkAndEncode(Texture2D source)
+    {
+        int targetHeight = Mathf.FloorToInt(source.height * imageShrinkingRatio);
+        int targetWidth = Mathf.FloorToInt(source.width * imageShrinkingRatio);
+        Texture2D scaledTex = new Texture2D(targetWidth, targetHeight, source.format, true);
+        Color[] pixelsColorArray = scaledTex.GetPixels(0);
+        float xRatio = ((float)1 / source.width) * ((float)source.width / targetWidth);
+        float yRatio = ((float)1 / source.height) * ((float)source.height / targetHeight);
+        for (int px = 0; px < pixelsColorArray.Length; px++)
+        {
+            pixelsColorArray[px] = source.GetPixelBilinear(xRatio * ((float)px % targetWidth), yRatio * ((float)Mathf.Floor(px / targetWidth)));
+        }
+        scaledTex.SetPixels(pixelsColorArray, 0);
+        scaledTex.Apply();
+        byte[] bytes = scaledTex.EncodeToJPG();
+        Destroy(scaledTex);
+        return bytes;
+    }
+}
+
+[DataContract]
+public class FaceDetectionResponse
+{
+    [DataMember]
+    public bool success;
+    [DataMember]
+    public float server_processing_time;
+    [DataMember]
+    public int[][] rects;
+}
+
+[DataContract]
+public class ObjectDetectionResponse
+{
+    [DataMember]
+    public bool success;
+    [DataMember]
+    public float server_processing_time;
+    [DataMember]
+    public bool gpu_support;
+    [DataMember]
+    public @Object[] objects;
+}
+
+[DataContract]
+public class @Object
+{
+    [DataMember]
+    public int[] rect;
+    [DataMember]
+    public string @class;
+    [DataMember]
+    public float confidence;
+}
+
+[DataContract]
```

Comment:
We may want to expose something this generic in the SDK itself.

Reply:
This will be super helpful for Indie Developers (90% of Unity Developer)